src/Core/src/Platform/Android/MauiSwipeView.cs

```diff
@@ -115,31 +115,36 @@ bool ShouldInterceptTouch(MotionEvent? e)
 
 		bool ShouldInterceptScrollChildrenTouch(SwipeDirection swipeDirection)
 		{
-			if (!(_contentView is ViewGroup viewGroup) || _initialPoint == null)
+			if (_contentView is null || _initialPoint is null)
 				return false;
 
-			int x = (int)(_initialPoint.X * _density);
-			int y = (int)(_initialPoint.Y * _density);
+			var viewGroup = _contentView as ViewGroup;
 
-			bool isHorizontal = swipeDirection == SwipeDirection.Left || swipeDirection == SwipeDirection.Right;
-
-			for (int i = 0; i < viewGroup.ChildCount; i++)
+			if (viewGroup is not null)
 			{
-				var child = viewGroup.GetChildAt(i);
+				int x = (int)(_initialPoint.X * _density);
+				int y = (int)(_initialPoint.Y * _density);
+
+				bool isHorizontal = swipeDirection == SwipeDirection.Left || swipeDirection == SwipeDirection.Right;
 
-				if (child != null && IsViewInBounds(child, x, y))
+				for (int i = 0; i < viewGroup.ChildCount; i++)
```

Comment:
'i' is just used to access the index, why don't we use a foreach in the collection? If we are worried about the performance of foreach vs for, we have bigger problems.

Reply:
Is mostly used to get the child using the `GetChildAt` method.